core-bundle/src/Resources/contao/library/Contao/Config.php

```diff
@@ -77,6 +77,34 @@ class Config
 	 */
 	protected $strRootDir;
 
+	private static $arrDeprecatedMap = array
+	(
+		'dbHost'           => 'database_host',
+		'dbPort'           => 'database_port',
+		'dbUser'           => 'database_user',
+		'dbPass'           => 'database_password',
+		'dbDatabase'       => 'database_name',
+		'smtpHost'         => 'mailer_host',
+		'smtpUser'         => 'mailer_user',
+		'smtpPass'         => 'mailer_password',
+		'smtpPort'         => 'mailer_port',
+		'smtpEnc'          => 'mailer_encryption',
+		'addLanguageToUrl' => 'contao.prepend_locale',
+		'urlSuffix'        => 'contao.url_suffix',
+		'uploadPath'       => 'contao.upload_path',
+		'editableFiles'    => 'contao.editable_files',
+		'debugMode'        => 'kernel.debug',
+		'characterSet'     => 'kernel.charset',
+		'enableSearch'     => 'contao.search.default_indexer.enable',
+		'indexProtected'   => 'contao.search.index_protected',
+	);
+
+	private static $arrDeprecated = array
```

Comment:
Why did you not add these two to `$arrDeprecatedMap`?

Reply:
Because `$arrDeprecatedMap` ist used on line 513 and is a direct mapping (without modifications).

Reply:
And `$GLOBALS['TL_CONFIG']['jpgQuality']` is different from `contao.image.imagine_options.jpeg_quality`?

Reply:
Yes. `contao.image.imagine_options` is the parameter that is an array. The parameter `contao.image.imagine_options.jpeg_quality` does not exist.

Reply:
We could rename to `contao.image.imagine_options[jpeg_quality]` to make the deprecation message more clear.

Reply:
Good idea. 👍🏻

Reply:
Done in e5ee6ee6880b0ca6ae12c35d9ed8398c7197d9ad